tests/sentry/middleware/test_ratelimit_middleware.py

```diff
@@ -44,3 +46,27 @@ def test_negative_rate_limit_check(self, default_rate_limit_mock):
         time.sleep(1)
         self.middleware.process_view(request, self.view, [], {})
         assert not request.will_be_rate_limited
+
+    def test_get_rate_limit_key(self):
+        # Mock view class
+        class OrganizationGroupIndexEndpoint(Endpoint):
+            pass
+
+        self.view = OrganizationGroupIndexEndpoint
+        request = self.factory.get("/")
+        request.session = {}
+        request.user = self.user
+        request.access = from_request(request, self.organization)
+
+        assert (
+            get_rate_limit_key(self.view, request)
+            == "ip:OrganizationGroupIndexEndpoint:GET:127.0.0.1"
```

Comment:
Will this break for IPv6 if we try to split on `:`?

Reply:
Shouldn't be an issue. The key is immediately hashed and used for redis. No information is extracted from it. 
However if we need to extract info for whatever reason, we can do `.split(maxsplit=3)` and then the IP (regardless of IPv4 or IPv6) will be in the last index of the split key.